src/plugins/search/search.js

```diff
@@ -98,6 +98,20 @@ export function genIndex(path, content = '', router, depth) {
         slug = router.toURL(path, { id: slugify(escapeHtml(text)) });
       }
 
+      // fix 404
+      let pathname = location.pathname;
+      let flag = false;
+      while (slug.startsWith('#/../') && pathname.lastIndexOf('/') > 1) {
+        flag = true;
+        if (pathname.endsWith('/')) {
+          pathname = pathname.substring(0, pathname.length - 1);
+        }
+        pathname = pathname.substring(0, pathname.lastIndexOf('/'));
+
+        slug = '#/' + slug.substring(5);
+      }
+      slug = pathname + (flag ? '/' : '') + slug;
+
```

Comment:
What does those changes for ?  could you plz give a sample and add verify test cases.

Reply:
When my website has multiple paths, it can prevent 404 error.
For example:
`http://xxxxxx/A/`
`http://xxxxxx/B/`
When searching on page A and accessing page B, 404 will appear.

Reply:
> When my website has multiple paths, it can prevent 404 error. For example: `http://xxxxxx/A/` `http://xxxxxx/B/` When searching on page A and accessing page B, 404 will appear.

I see, maybe u can take a look on this `namespace` configs [here](https://docsify.js.org/#/plugins?id=full-text-search).

Reply:
Should I configure it like this?

```json
{
    search: [
        '/A',
        '/B'
    ]
}
```

Reply:
I guess you need distinguish those namespace for different paths (sites).
```js


      // Use different indexes for path prefixes (namespaces).
      // NOTE: Only works in 'auto' mode.
      //
      // When initialiazing an index, we look for the first path from the sidebar.
      // If it matches the prefix from the list, we switch to the corresponding index.
      pathNamespaces: ['/zh-cn', '/ru-ru', '/ru-ru/v1'],
```

Reply:
Ok, I will try it.